internal/upload/koji/koji.go

```diff
@@ -117,34 +151,36 @@ func Login(server, user, password string, transport http.RoundTripper) (*Koji, e
 	}
 
 	args := []interface{}{user, password}
-	var reply struct {
-		SessionID  int64  `xmlrpc:"session-id"`
-		SessionKey string `xmlrpc:"session-key"`
-	}
+	var reply loginReply
 	err = loginClient.Call("login", args, &reply)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create the final xmlrpc client with our custom RoundTripper handling
-	// sessionID, sessionKey and callnum
-	kojiTransport := &Transport{
-		sessionID:  reply.SessionID,
-		sessionKey: reply.SessionKey,
-		callnum:    0,
-		transport:  transport,
+	return newKoji(server, transport, reply)
+}
+
+func GSSAPILogin(server string, credentials *GSSAPICredentials, transport http.RoundTripper) (*Koji, error) {
```

Comment:
Further to my previous comment, maybe name the constructors `NewFromPlain()` and `NewFromGSSAPI()`, or whatever the canonical names of these things are ;)

Reply:
Fixed